examples/room-manager/room_service.py

```diff
@@ -0,0 +1,166 @@
+from argparse import Namespace
+from dataclasses import dataclass
+from logging import Logger
+from typing import List
+
+import betterproto
+
+from fishjam import FishjamClient, PeerOptions, Room, RoomOptions
+from fishjam.events import ServerMessagePeerCrashed as PeerCrashed
+from fishjam.events import ServerMessagePeerDeleted as PeerDeleted
+from fishjam.events import ServerMessageRoomCrashed as RoomCrashed
+from fishjam.events import ServerMessageRoomDeleted as RoomDeleted
+
+
+@dataclass
+class Info:
+    id: str
+    name: str
+
+
+@dataclass
+class PeerAccess:
+    room: Info
+    peer: Info
+    peer_token: str
+    websocket_url: str
+
+
+class RoomService:
+    def __init__(self, args: Namespace, logger: Logger):
+        self.fishjam_client = FishjamClient(
+            fishjam_url=args.fishjam_url,
+            management_token=args.management_token,
+        )
+        self.websocket_url = args.fishjam_url.replace("http", "ws")
+        self.room_name_to_room_id: dict[str, str] = {}
+        self.peer_name_to_access: dict[str, PeerAccess] = {}
+        self.logger = logger
+        self.config = args
+
+    def get_peer_access(self, room_name: str, username: str) -> PeerAccess:
+        room = self.__find_or_create_room(room_name)
+        peer_access = self.peer_name_to_access.get(username)
+        peer_in_room = self.__is_in_room(room, peer_access)
+
+        self.logger.info(
+            "Got room: %s", {"name": room_name, "id": room.id, "peers": room.peers}
+        )
+
+        if not peer_access or not peer_in_room:
+            return self.__create_peer(room_name, username)
+
+        self.logger.info("Peer and room exist: %s, %s", username, room_name)
```

Comment:
`nitpick:` If room does exist we log `room exists` in the `__find_or_create_room()` and then we log `peer and room exist`. So we inform about room existence twice.